### ISSUE TITLE
Add option to disable VP9 codec

### ISSUE THREAD

**Issue by matejdro** [anxdpanic/plugin.video.youtube#324]
### What Problem Does This Solve? 

Nvidia Shield 2019 (the tube version) does not have a hardware decoder for VP9 videos.

Up until recently, this was not the issue as only HDR videos were in the VP9 (so just disabling the HDR effectively disabled VP9 codec). However, recently it seems like Youtube also started to offer regular 4K videos in VP9.

Plugin seems to prefer to select VP9 video over the h265 4K stream, which causes a very choppy playback on the Shield 2019.

Current workaround is to limit the quality to 1080p, but this will also prevent 4K streams from videos that have H265 4K stream available.

### Suggest a Solution (optional)

Add an option to the MPEG-DASH menu (next to the HDR toggle) to enable/disable streams with VP9 codec. Plugin should fallback to H265 4K (if available, otherwise just pick a best non-VP9 quality available).

**Comment by matejdro** [comment]
It appears I was wrong. The tube supports VP9, stuttering was apparently caused by something else.

**Comment by peat-psuwit** [comment]
Hello. I'm also affected by this issue. Although, in this case the hardware is the original Raspberry Pi B+ [1], which doesn't support H/W accelerated VP9 and is too weak to handle even 144p video in CPU.

Since it seems like this plugin already filter MPD passed to inputstream.adaptive, maybe we can take advantage of it here.

Can we re-open this issue?

[1] XBian still provides updates for it - I'm getting Kodi v20 Nexus even!

**Comment by Isayso** [comment]
I support this option. RPI 3/3+ is also not able to play vp9 videos.

PS: I noticed that xbian has problem with vp9 codec.

**Comment by MoojMidge** [comment]
What is the latest version of Kodi that is officially provided/supported by the main RPI distributions? Is Kodi v20 widely available?

**Comment by Isayso** [comment]
> What is the latest version of Kodi that is officially provided/supported by the main RPI distributions? Is Kodi v20 widely available?

That is not an easy answer. The RPI 2/3 seems to be neglected now. I have several RPI3/3+ and jiggle with 3 versions.

LibreElec, xbian and plain Debian with Kodi.

Libreelec 11 has 20.1 and lot's of problems on RPI3/3+, frequent crashes, no reliable 60Hz, audio out of sync/distortions every 10 minutes, issues with some Addons   

xbian has 20.2 now, much more stable, 60 Hz works, but vp9 does not work and is very slow in general, CEC reaction time slow compared to libreelec. Libreelec on RPI3 is faster than xbian on 3+ overclocked.   On my RPI3 I still use Libreelec 10 with 19.3, which has fewer problems. 

Debian with Kodi is 19.4 (or 5?) and has the same issues as Libreelec. 

Since Matrix it is a mess for RPI's, sorry

**Comment by MoojMidge** [comment]
Clear as mud.

There is some code that seperates the video features from the resolution. It currenly only work on Kodi 20+, but should work on Kodi 19 as well without too many changes except for the ListItem InfoTagVideo changes.

![image](https://github.com/anxdpanic/plugin.video.youtube/assets/56883549/334c91df-23dd-4a0f-a904-6deba8fc88af)

![image](https://github.com/anxdpanic/plugin.video.youtube/assets/56883549/106d28f3-232c-4e75-9043-edfd72ba64f8)

For Kodi 19, a quick workaround in the meantime is to change these lines:
https://github.com/anxdpanic/plugin.video.youtube/blob/b07f541b04d71731ebf69cbb67eb638d0f2bd7fe/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_context.py#L312C13-L313C30
from
```
            'vp9': '2.3.14',
            'vp9.2': '2.3.14',
```
to
```
            'vp9': None,
            'vp9.2': None,
```

**Comment by MoojMidge** [comment]
Fixed by #506
